core/vm/interpreter.go

```diff
@@ -133,13 +137,23 @@ func (in *Interpreter) Run(snapshot int, contract *Contract, input []byte) (ret
 	for atomic.LoadInt32(&in.evm.abort) == 0 {
 		// Get the memory location of pc
 		op = contract.GetOp(pc)
+		logged = false
 
 		// get the operation from the jump table matching the opcode
 		operation := in.cfg.JumpTable[op]
 		if err := in.enforceRestrictions(op, operation, stack); err != nil {
 			return nil, err
 		}
 
+		if in.cfg.Debug {
+			pcCopy = uint64(pc)
+			gasCopy = uint64(contract.Gas)
+			stackCopy = newstack()
+			for _, val := range stack.data {
+				stackCopy.push(val)
```

Comment:
What's the intent here? The `stackCopy` will contain the same `bigint` objects as the original stack, so if any of those are modified (e.g. by being popped, added to intpool, and then reused), they will change also in the `stackCopy`.